### PR TITLE
COMMERCE-10449 - Reject Cookies to be handled with Compare Products

### PR DESCRIPTION
**Pull request by ilzamcmed**
[COMMERCE-10741](https://issues.liferay.com/browse/COMMERCE-10741)

### CODE REVIEW

**Comment by ilzamcmed** [comment]
This PR is ready for review
cc: @alessiorendina

**Comment by ilzamcmed** [comment]
This PR is ready for review
cc: @andrea-ale-sbarra

**Comment by ilzamcmed** [comment]
ci:test:sf

**Comment by ilzamcmed** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_compare/MiniCompare.js [hunk 1]
```diff
@@ -197,9 +184,15 @@ function MiniCompare(props) {
 					);
 				})}
 
-			<a className="btn btn-primary" href={props.compareProductsURL}>
-				{Liferay.Language.get('compare')}
-			</a>
+			{renderButton}
```
Comment (interaminense) [inline modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_compare/MiniCompare.js:188]: Could you render button directly here instead?

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_compare/_mini_compare.scss [hunk 2]
```diff
@@ -109,3 +109,20 @@
...
 	}
 }
+
+.cookies-modal {
+	.modal-dialog {
+		max-width: 630px;
+
+		.alert {
+			opacity: 0.7;
+		}
```
Comment (interaminense) [inline modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_compare/_mini_compare.scss:119]: Please, break line.

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_compare/ConfirmationCookiesModal.js [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+		onClose();
+		onDeclineFunctionalCookie();
+	};
+
+	const handleAcceptAll = () => {
+		handleSetCookies({
+			functionalCookies: true,
+			performanceCookies: true,
+			personalizationCookies: true,
+		});
```
Comment (interaminense) [inline modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_compare/ConfirmationCookiesModal.js:116]: Please, break line.

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_compare/ConfirmationCookiesModal.js [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+		setFunctionalCookies(true);
+		setPerformanceCookies(true);
+		setPersonalizationCookies(true);
```
Comment (interaminense) [inline modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_compare/ConfirmationCookiesModal.js:119]: Please, break line.

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_compare/ConfirmationCookiesModal.js [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+		alertCookies(
+			'success',
+			Liferay.Language.get('commerce-cookies-accepted'),
+			Liferay.Language.get('commerce-cookies-accepted-description')
+		);
```
Comment (interaminense) [inline modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_compare/ConfirmationCookiesModal.js:124]: Please, break line.

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_compare/ConfirmationCookiesModal.js [hunk 6]
```diff
@@ -0,0 +1,271 @@
...
+		}
+		onClose();
+	};
+
+	const handleDeclineAll = () => {
+		handleSetCookies({
+			functionalCookies: false,
+			performanceCookies: false,
+			personalizationCookies: false,
+		});
```
Comment (interaminense) [inline modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_compare/ConfirmationCookiesModal.js:98]: Please, break line.

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_compare/ConfirmationCookiesModal.js [hunk 7]
```diff
@@ -0,0 +1,271 @@
...
+		setFunctionalCookies(false);
+		setPerformanceCookies(false);
+		setPersonalizationCookies(false);
```
Comment (interaminense) [inline modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_compare/ConfirmationCookiesModal.js:101]: Please, break line.

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_compare/ConfirmationCookiesModal.js [hunk 8]
```diff
@@ -0,0 +1,271 @@
...
+			COOKIE_TYPES.PERSONALIZATION
+		);
+	};
+
+	const handleConfirm = () => {
+		handleSetCookies({
+			functionalCookies,
+			performanceCookies,
+			personalizationCookies,
+		});
```
Comment (interaminense) [inline modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_compare/ConfirmationCookiesModal.js:74]: Please, break line.

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_compare/ConfirmationCookiesModal.js [hunk 9]
```diff
@@ -0,0 +1,271 @@
...
+				Liferay.Language.get('commerce-cookies-rejected-description')
+			);
+			onDeclineFunctionalCookie();
+		} else {
+			alertCookies(
+				'success',
+				Liferay.Language.get('commerce-cookies-accepted'),
+				Liferay.Language.get('commerce-cookies-accepted-description')
+			);
+		}
```
Comment (interaminense) [inline modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_compare/ConfirmationCookiesModal.js:89]: Please, break line.

**Review by interaminense — CHANGES_REQUESTED** [review]
Thanks @ilzamcmed. I left few comments, could you take a look?

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties [hunk 10]
```diff
@@ -20005,4 +20005,9 @@ zip=Zip
 zip-postal=Zip/Postal
 zoom-in=Zoom In
 zoom-out=Zoom Out
-zoom-to-fit=Zoom to Fit
\ No newline at end of file
+zoom-to-fit=Zoom to Fit
+commerce-cookies-rejected=Cookies Rejected:
+commerce-cookies-rejected-description=Without cookies acceptance it’s not possibile to use compare function
```
Comment (interaminense) [inline modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:20010]: I think `it’s` is differente this `it's`

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties [hunk 11]
```diff
@@ -20005,4 +20005,9 @@ zip=Zip
...
+commerce-cookies-accepted=Cookies Accepted: 
+commerce-cookies-accepted-description=It’s now possibile to use the compare function
```
Comment (interaminense) [inline modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:20012]: The same thing here.

**Review by interaminense — CHANGES_REQUESTED** [review]

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_compare/ConfirmationCookiesModal.js [hunk 12]
```diff
@@ -0,0 +1,271 @@
...
+	const alertCookies = (alertType, alertTitle, alertMessage) => {
+		Liferay.Util.openToast({
+			message: alertMessage,
+			title: alertTitle,
+			toastProps: {
+				autoClose: 5000,
+			},
+			type: alertType,
+		});
+	};
```
[the remark this answers is not in the document]
Comment (ilzamcmed) [reply modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_compare/ConfirmationCookiesModal.js:47]: Fixed

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_compare/ConfirmationCookiesModal.js [hunk 13]
```diff
@@ -0,0 +1,271 @@
...
+						</Text>
+					) : (
+						<ClayToggle onToggle={onToggle} toggled={toggled} />
+					)}
+				</div>
+
+				<Text as="p">{description}</Text>
+			</div>
+		);
+	};
```
[the remark this answers is not in the document]
Comment (ilzamcmed) [reply modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_compare/ConfirmationCookiesModal.js:154]: Fixed

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_compare/ConfirmationCookiesModal.js [hunk 14]
```diff
@@ -0,0 +1,271 @@
...
+			'success',
+			Liferay.Language.get('commerce-cookies-accepted'),
+			Liferay.Language.get('commerce-cookies-accepted-description')
+		);
+		onClose();
+	};
+
+	const CookieComponent = ({
+		description,
+		necessaryCookie = false,
```
[the remark this answers is not in the document]
Comment (ilzamcmed) [reply modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_compare/ConfirmationCookiesModal.js:130]: Fixed

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_compare/MiniCompare.js (hunk 1 above)
Comment (ilzamcmed) [reply modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_compare/MiniCompare.js:188]: Fixed

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_compare/_mini_compare.scss (hunk 2 above)
Comment (ilzamcmed) [reply modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_compare/_mini_compare.scss:119]: Fixed

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_compare/ConfirmationCookiesModal.js (hunk 3 above)
Comment (ilzamcmed) [reply modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_compare/ConfirmationCookiesModal.js:116]: Fixed

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_compare/ConfirmationCookiesModal.js (hunk 4 above)
Comment (ilzamcmed) [reply modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_compare/ConfirmationCookiesModal.js:119]: Fixed

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_compare/ConfirmationCookiesModal.js (hunk 5 above)
Comment (ilzamcmed) [reply modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_compare/ConfirmationCookiesModal.js:124]: Fixed

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_compare/ConfirmationCookiesModal.js (hunk 6 above)
Comment (ilzamcmed) [reply modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_compare/ConfirmationCookiesModal.js:98]: Fixed

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_compare/ConfirmationCookiesModal.js (hunk 7 above)
Comment (ilzamcmed) [reply modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_compare/ConfirmationCookiesModal.js:101]: Fixed

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_compare/ConfirmationCookiesModal.js (hunk 8 above)
Comment (ilzamcmed) [reply modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_compare/ConfirmationCookiesModal.js:74]: Fixed

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_compare/ConfirmationCookiesModal.js (hunk 9 above)
Comment (ilzamcmed) [reply modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_compare/ConfirmationCookiesModal.js:89]: Fixed

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties (hunk 10 above)
Comment (ilzamcmed) [reply modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:20010]: After `buildLang`, it corrected to:

> It is 

Should I keep it?

Comment (interaminense) [reply modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:20010]: Oh sorry, I didn't see it. Thanks Ilza!

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties (hunk 11 above)
Comment (ilzamcmed) [reply modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:20012]: Fixed

Comment (ilzamcmed) [reply modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties]: `buildLang` keeps correcting the text to 

> It is

**Review by andrea-ale-sbarra — COMMENTED** [review]
@ilzamcmed @interaminense looks good to me, but I don't see the fixes made for the changing requests, can you @ilzamcmed point me where these changes are?

From the functional point of view looks to me that this is doing the job  👍 

Thank you!

**Comment by ilzamcmed** [comment]
Hey @andrea-ale-sbarra  you can find the fixes in this commit:  https://github.com/liferay-commerce/liferay-portal/pull/3405/commits/f3c4ded99c1d9af2ff4b92c61959c80a72b58554

**Review by interaminense — APPROVED** [review]
LGTM! Thanks @ilzamcmed :fire:

**Review by andrea-ale-sbarra — APPROVED** [review]

**Comment by victoroscuro** [comment]
ci:test:relevant

**Comment by ilzamcmed** [comment]
ci:test:relevant

**Comment by ilzamcmed** [comment]
ci:test:commerce-functional

**Comment by ilzamcmed** [comment]
ci:test:commerce-functional

**Comment by ilzamcmed** [comment]
ci:test:relevant

**Comment by ilzamcmed** [comment]
ci:test:relevant

**Comment by ilzamcmed** [comment]
ci:test:commerce-functional

**Comment by ilzamcmed** [comment]
Developed as part of https://issues.liferay.com/browse/COMMERCE-10535

**Pull request closed without merging**